bundle/src/main/java/com/adobe/acs/commons/remoteassets/impl/RemoteAssetsNodeSyncImpl.java

```diff
@@ -0,0 +1,606 @@
+
+                    if (":".concat(JcrConstants.JCR_DATA).equals(key)) {
+                        setJcrDataProperty(parentNode, json.getString(JcrConstants.JCR_LASTMODIFIED));
+                    } else if (key.startsWith(":")) {
+                        // Skip binary properties, since they do not come across in JSON
+                        continue;
+                    } else if (parentNode.hasProperty(key) && parentNode.getProperty(key).getString().equals(value)) {
+                        // Skipping due to the property already existing and being equal
+                        continue;
+                    } else if (PROTECTED_PROPERTIES.contains(key)) {
+                        // Skipping due to the property being unmodifiable.
+                        continue;
+                    } else {
+                        setProperty(value, key, parentNode);
+                    }
+                } catch (RepositoryException re) {
+                    LOG.warn("Repository exception thrown. Skipping '{}' single property for node '{}'.", key, parentNode.getPath());
+                }
+            }
+        }
+    }
+
+    /**
+     * Set mixins property for the property's node.
+     * @param json JSONObject
+     * @param key String
+     * @param node Node
+     * @throws JSONException exception
+     * @throws RepositoryException exception
+     */
+    private void setMixinsProperty(final JSONObject json, final String key, final Node node) throws JSONException, RepositoryException {
+        JSONArray mixins = (JSONArray) json.get(key);
+        for (int i = 0; i < mixins.length(); i++) {
+            node.addMixin(mixins.getString(i));
+            if (LOG.isDebugEnabled()) {
+                LOG.debug("Mixin '{}' added for node '{}'.", mixins.getString(i), node.getPath());
+            }
+        }
+    }
+
+    /**
+     * Set tags property for the property's node.
+     * @param json JSONObject
+     * @param key String
+     * @param node Node
+     * @throws JSONException exception
+     * @throws RepositoryException exception
+     */
+    private void setTagsProperty(final JSONObject json, final String key, final Node node) throws JSONException, RepositoryException {
+        TagManager tagManager = this.resourceResolver.adaptTo(TagManager.class);
+        JSONArray tags = (JSONArray) json.get(key);
+        ArrayList<Tag> tagList = new ArrayList<>();
+
+        for (int i = 0; i < tags.length(); i++) {
+            Tag tag = tagManager.resolve(tags.getString(i));
+            if (tag == null) {
+                LOG.warn("Tag '{}' could not be found. Skipping tag for node '{}'.", tags.getString(i), node.getPath());
+                continue;
+            }
+
+            tagList.add(tag);
+        }
+
+        if (tagList.size() > 0) {
+            Resource parentResource = this.resourceResolver.getResource(node.getPath());
+            tagManager.setTags(parentResource, tagList.toArray(new Tag[0]));
```

Comment:
Technically I think its more efficient if you use `new Tag[tagList.size()]` instead of `new Tag[0]` b/c otherwise the internal implementatino had to create a new array that is big enough for the list.